### PR TITLE
Master forecast clean jem

### PR DESCRIPTION
**Pull request by jem-odoo**
Description of the issue/feature this PR addresses:

Current behavior before PR:

Desired behavior after PR is merged:

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: odoo/tools/misc.py [hunk 1]
```diff
@@ -1196,6 +1198,72 @@ def format_date(env, value, lang_code=False, date_format=False):
...
+     : param {str, datetime} value: datetime to format either in string or in datetime
+    """
+    if not value:
+        return ''
+    if isinstance(value, str):
+        timestamp = odoo.fields.Datetime.from_string(value)
+    else:
+        timestamp = value
+
+    tz_name = tz or env.user.sudo().tz or 'UTC'
```
Comment (tde-banana-odoo) [inline odoo/tools/misc.py]: Side question: isn't the user on environment already sudo-ed ? We could gain 20 000 queries ...

File: odoo/tools/misc.py [hunk 2]
```diff
@@ -1196,6 +1198,72 @@ def format_date(env, value, lang_code=False, date_format=False):
 
     return babel.dates.format_date(value, format=date_format, locale=locale)
 
+
+def format_datetime(env, value, tz=False, dt_format='medium', lang_code=False):
+    """ Formats the datetime in a given format.
+
+     : param {str, datetime} value: datetime to format either in string or in datetime
```
Comment (tde-banana-odoo) [inline odoo/tools/misc.py]: Could be great to explain a bit the parameters :) .

**Comment by tde-banana-odoo** [comment]
Side note: format_datetime has been improved, not just moved. Could be great to tell it in commit message (and write tests someday ?).

File: odoo/tools/misc.py (hunk 1 above)
Comment (jem-odoo) [reply odoo/tools/misc.py]: Yes indeed ! So this is not required !

**Comment by jem-odoo** [comment]
@robodoo rebase-ff r+

**Pull request closed without merging**